### PR TITLE
Add a verbose logging option to mimic

### LINKED ISSUE #40: content-type should be application/json and not twisted's unfortunate default of text/html
There really ought to be a decorator which does both this and JSON serialization.

### LINKED ISSUE #202: stop setting Request.defaultContentType
I think there was some confusion around this being specific to an individual file, rather than a temporary (and gross) hack to make a specific change.  We should be careful to set the content type in each response.  If copy/pasting the lines of code to do `json.dumps` and `setHeader` are tedious, we should make a decorator that combines it with `route`, perhaps, to avoid duplication, but we shouldn't manipulate global state.

### LINKED ISSUE #362: Verbose mode which logs requests and responses
Log the headers and body of every incoming request, as well as the headers, status code, and body of every response.

### PR DESCRIPTION
**Pull request by cyli**
So it logs all incoming requests and outgoing responses.

Fixes #362.
Fixes #202.
Addresses part of #40.

### CODE REVIEW

**Comment by cyli** [comment]
Hmm... the coverage diff it calculated is wrong.   I think codecov was having trouble with the coverage diff because of the merge from master?  It only seemed to display my last commit.  Rebasing to see if that fixes it.

File: mimic/resource.py [hunk 1]
```diff
@@ -130,3 +147,70 @@ def get_service_resource(self, request, service_id, region_name):
             # workaround for https://github.com/twisted/klein/issues/56
             return NoResource()
         return service_object
+
+
+class MimicRequest(Request):
```
Comment (lekhajee) [inline mimic/resource.py]: Per @glyph's observation, `MimicRequest` should be inheriting `object` as against `MimicLoggingRequest`

File: mimic/resource.py [hunk 2]
```diff
@@ -1,10 +1,27 @@
...
 import json
 
+from io import StringIO
+
+from six import text_type
+
 from twisted.web.resource import NoResource
+from twisted.web.server import Request, Site
+
+try:
```
Comment (lekhajee) [inline mimic/resource.py]: Could we please add a comment to explain this, so we know to get rid of it when we make decisions on versions supported and such.

**Comment by lekhajee** [comment]
+1 lgtm modulo the nits

**Comment by lekhajee** [comment]
@cyli: merging this, could you please do a follow up PR for the nits. Thanks !

**Comment by cyli** [comment]
@lekhajee Will do, thanks!

**Pull request merged**